app/src/main/java/com/worldsnas/starplayer/ExoPlayerService.kt

```diff
@@ -0,0 +1,425 @@
+package com.worldsnas.starplayer
+
+import android.app.Notification
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.app.Service
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import android.os.Build
+import android.os.IBinder
+import android.text.TextUtils
+import androidx.core.app.NotificationCompat
+import com.google.android.exoplayer2.*
+import com.google.android.exoplayer2.source.*
+import com.google.android.exoplayer2.trackselection.TrackSelectionArray
+import com.google.android.exoplayer2.upstream.DefaultHttpDataSourceFactory
+import com.worldsnas.starplayer.model.Music
+import java.util.*
+
+class ExoPlayerService : Service(), Player.EventListener {
+
+
+    lateinit var mediaSource: ConcatenatingMediaSource
+
+    lateinit var player: SimpleExoPlayer
+
+    private var currentWindow = 0
+    private var playbackPosition: Long = 0
+    private var musicList: ArrayList<Music?>? = null
+    private var currentMusicId: Int = 0
+
+
+    private fun reInit() {
+        playbackPosition = 0
+        prepareCurrentWindow(musicList, currentMusicId)
+        initializePlayer()
+    }
+
+
+    override fun onDestroy() {
+        super.onDestroy()
+        releasePlayer()
+    }
+
+
+    private fun play(trackModel: Music?) {
+        val trackModels = ArrayList<Music?>()
+        trackModels.add(trackModel)
+        play(trackModels)
+    }
+
+    private fun play(trackModels: ArrayList<Music?>?) {
+        reInit()
```

Comment:
benazaram behtare moghe sakhte shodan e service yekbar faghat player ro besazim
injori dari hardafe ye player jadid misazi

Reply:
albateh to injam faghat yebar sakhte mishe

Reply:
ok